### ISSUE TITLE
[5.x] v5.0.0-RC5: Specifying kubernetes_version does not update k8s in node-pools

### ISSUE THREAD

**Issue by rodrigc** [oracle-terraform-modules/terraform-oci-oke#828]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version and Provider Version

```
terraform -v
Terraform v1.4.4
on darwin_amd64
+ provider registry.terraform.io/hashicorp/cloudinit v2.3.2
+ provider registry.terraform.io/hashicorp/helm v2.11.0
+ provider registry.terraform.io/hashicorp/http v3.4.0
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/hashicorp/random v3.5.1
+ provider registry.terraform.io/hashicorp/time v0.9.1
+ provider registry.terraform.io/oracle/oci v4.119.0
```

### Affected Resource(s)

[oci_containerengine_node_pool](https://docs.oracle.com/en-us/iaas/tools/terraform-provider-oci/5.12/docs/r/containerengine_node_pool.html)

### Terraform Configuration Files

```hcl
module "oke" {
  source = "github.com/oracle-terraform-modules/terraform-oci-oke?ref=v5.0.0-RC5"

  kubernetes_version         = "v1.26.2"
  cluster_name               = local.name
  control_plane_is_public    = false
  cluster_type               = "enhanced"
  cluster_autoscaler_install = true

  compartment_id         = module.oke_prereqs.oke_compartment_id
  network_compartment_id = module.oke_prereqs.oke_compartment_id
  worker_compartment_id  = module.oke_prereqs.oke_compartment_id
  tenancy_id             = local.tenancy_id

  ssh_private_key_path = var.ssh_private_key_path
  ssh_public_key_path  = var.ssh_public_key_path

  home_region = local.home_region
  region      = var.region

  // node pool config
  worker_pools            = var.worker_pools
  worker_image_id         = "none"
  worker_image_type       = "platform"
  worker_image_os         = "Oracle Linux"
  worker_image_os_version = "8"

  // leaving this disabled for now, previously it didn't actually enable
  // kms secrets like it should, will have to go back and fix that if
  // we want kms encryption (we still have AES encryption).
  //cluster_kms_key_id = module.oke_prereqs.oke_kms_key_id

  create_operator       = true
  create_bastion        = true

  create_iam_resources       = true
  create_iam_operator_policy = "always"

  providers = {
    oci.home = oci.home
  }
}
```

### Problem

1.  I originally created a Kubernetes cluster, and the version was v1.25.4.
2.  I then updated the Terraform to specify `kubernetes_version = "v1.26.2"`
3.  I did a `terraform apply`
4. The control plane of the Kubernetes cluster was upgraded to `v1.26.2`
5.  However, the Kubernetes versioin in the node pools were not updated and is still at `v1.25.4`

6.  I queried the node pool:
```
oci ce node-pool get --node-pool-id ocid1.nodepool.oc1.region.xxxxxx
```

and see:

```json
 {
        "availability-domain": "xJwU:US-ASHBURN-AD-1",
        "defined-tags": null,
        "fault-domain": "FAULT-DOMAIN-3",
        "freeform-tags": null,
        "id": "ocid1.instance.oc1.region.XXX",
        "kubernetes-version": "v1.25.4",
```

There needs to be a way to specify the `kubernetes_version`, and have it update the node pool metadata in OCI.

**Comment by rodrigc** [comment]
This line:
https://github.com/oracle-terraform-modules/terraform-oci-oke/blob/5.x/modules/workers/nodepools.tf#L105

ignores changes to the `kubernetes_version` for node pools.

There needs to be a way to upgrade the `kubernetes_version` in node pools, especially if the `kubernetes_version` is upgraded in the control plane.

**Comment by hyder** [comment]
Resolved in #830
